### ISSUE TITLE
@types/multer-gridfs-storage - Incorrect dependencies cause conflict with mongoose's bundled type definitions

### ISSUE THREAD

**Issue by samueldemoura** [DefinitelyTyped/DefinitelyTyped#52686]
- [x] I tried using the `@types/multer-gridfs-storage` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).
- [x] Authors: @devconcept

It seems mongoose now comes bundled with its own type definitions as of `mongoose@5.11.0` ([source](https://stackoverflow.com/a/65970740)). However, `@types/multer-gridfs-storage` still lists `@types/mongoose` as a dependency, which causes build failures due to conflicting definitions:
```
node_modules/@types/mongoose/index.d.ts:78:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: DocumentDefinition, FilterQuery, UpdateQuery, NativeDate, NativeError, Mongoose, SchemaTypes, STATES, connection, connections, models, mongo, version, CastError, ConnectionOptions, Collection, Connection, disconnected, connected, connecting, disconnecting, uninitialized, Error, QueryCursor, VirtualType, Schema, SchemaDefinition, SchemaTypeOpts, Subdocument, Array, DocumentArray, Buffer, ObjectId, ObjectIdConstructor, Decimal128, Map, Query, DocumentQuery, mquery, Aggregate, SchemaType, Promise, PromiseProvider, Model, Document, ModelUpdateOptions

78 declare module "mongoose" {
   ~~~~~~~

  node_modules/mongoose/index.d.ts:1:1
    1 declare module 'mongoose' {
      ~~~~~~~
    Conflicts are in this file.
```

Basically, the same error as https://github.com/DefinitelyTyped/DefinitelyTyped/issues/49950 but simply not installing the DefinitelyTyped types is not a solution in this case, since `@types/multer-gridfs-storage` forcibly installs `@types/mongoose`.

As a temporary workaround, running `rm -r node_modules/@types/mongoose` after `npm install` operations allows TypeScript to transpile the project as expected.

**Comment by devconcept** [comment]
The module was rewritten in typescript and now has it's own type definitions. The mongoose object is now not a direct reference but a mongoose-like object. Give the new version a try and close this issue if the solution works for you. The types distributed with this repo will be deprecated and removed in a few days.

**Comment by samueldemoura** [comment]
Apologies for the extremely delayed response, however I finally got a chance to test the new version and I can confirm everything is working as expected now. I'll be closing the issue.
